### PR TITLE
feat: [UI] Support moving glossary terms across glossary hierarchy #6601

### LINKED ISSUE #6601: We're starting using at 0.11.0 and realized that we cannot move Glossary Terms through the Glossary Tree. We have a big structure of terms so the mistakes of uncorrectly placed terms are guaranteed.
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

Closes #6601 
- Added functionality to move glossaryTerm under another glossaryTerm

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/225105527-2dbf6c51-1bac-4909-ad6c-c6db57be2f80.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx [hunk 1]
```diff
@@ -220,16 +236,61 @@ const GlossaryTermTab = ({
...
+        updatedGlossaryTerm
+      );
+
+      try {
+        await patchGlossaryTerm(movedGlossaryTerm.from?.id || '', jsonPatch);
+        fetchGlossaryTerm(
+          { glossary: glossaryId, parent: glossaryTermId },
+          false,
+          false
+        );
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:284]:
```suggestion
       await  fetchGlossaryTerm(
          { glossary: glossaryId, parent: glossaryTermId },
          false,
          false
        );
```

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx [hunk 2]
```diff
@@ -297,15 +358,29 @@ const GlossaryTermTab = ({
...
+              size="small"
+              onRow={(record, index) => {
+                const attr = {
+                  index,
+                  handleMoveRow,
+                  record,
+                };
+
+                return attr as DraggableBodyRowProps;
+              }}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:381]: Can we move this logic to the handler?

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx [hunk 3]
```diff
@@ -220,16 +236,61 @@ const GlossaryTermTab = ({
...
+    if (movedGlossaryTerm) {
+      setIsTableLoading(true);
+      const updatedGlossaryTerm = {
+        ...movedGlossaryTerm.from,
+        parent: {
+          fullyQualifiedName: movedGlossaryTerm.to.fullyQualifiedName,
+        },
+      };
+      const jsonPatch = compare(
+        movedGlossaryTerm.from as GlossaryTerm,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:274]: Do we need this type casting?

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:274]: correct, we can avoid let me have a look

**Review by Sachin-chaurasiya — APPROVED** [review]
Looks Good 👍

**Pull request merged**